src/couch_replicator_httpc.erl

```diff
@@ -98,6 +98,7 @@ send_ibrowse_req(#httpdb{headers = BaseHeaders} = HttpDb, Params) ->
         lists:ukeymerge(1, get_value(ibrowse_options, Params, []),
             HttpDb#httpdb.ibrowse_options)
     ],
+    backoff_before_request(Worker, HttpDb, Params),
```

Comment:
It seemed simpler from the code point of view to have one place to do it -- before the request, rather than having it in the success and failure. If a 429 happens, it updates the  interval, then we retries immediately.